nav2_behavior_tree/plugins/decorator/goal_updater_node.cpp

```diff
@@ -59,8 +59,13 @@ inline BT::NodeStatus GoalUpdater::tick()
 
   callback_group_executor_.spin_some();
 
-  if (rclcpp::Time(last_goal_received_.header.stamp) > rclcpp::Time(goal.header.stamp)) {
-    goal = last_goal_received_;
+  if (last_goal_received_.header.stamp != rclcpp::Time(0)) {
+    if (rclcpp::Time(last_goal_received_.header.stamp) > rclcpp::Time(goal.header.stamp)) {
+      goal = last_goal_received_;
+    }
+    else {
+      RCLCPP_WARN(node_->get_logger(), "Goal is older than the current one. Not updating");
```

Comment:
Add the timestamps here. I think this could be useful to ID problems with ROS clocks which would be immediately clear if we had them printed